### PR TITLE
Hyperparameter tuning to optimize the LightGBM and CatBoost models

### LINKED ISSUE #10: Hyperparameter tuning
In most of these algorithms, the basic parameters are taken to be default. I wish to add hyper-parameter tuning using Grid Search CV in SVR and KNN to see if the error reduces and Random Forest for experimental purposes.

I am a GGSoC Ext '24 Contributor. Kindly assign this issue to me for fixing.

Also, kindly add (gssoc-ext) and (hacktoberfest-accepted) as labels

### PR DESCRIPTION
**Pull request by SaurabhIndi**
fixes #10 
Implemented hyperparameter tuning using tools like GridSearchCV to optimize LightGBM and CatBoost models existing in the notebook.
1. Selected hyperparameters to tune: We will define a grid or range of hyperparameters for both LightGBM and CatBoost.
2. Implemented GridSearchCV: We will use these methods from sklearn.model_selection to find the optimal hyperparameters based on cross-validation.

### CODE REVIEW

**Review by Mayureshd-18 — APPROVED** [review]
@jvedsaqib Please validate the file structural changes.
@rohitinu6 Please verify the technical details.
Other than that I have checked. Looks good to me!
